railties/lib/rails/generators/rails/app/app_generator.rb

```diff
@@ -141,6 +141,7 @@ def test
     end
 
     def tmp
+      empty_directory_with_keep_file "tmp"
```

Comment:
I consider the mkdir_p a workaround for a "bug" we have. Ideally we shouldn't have to do it if we have this, but we didn't have that for old apps so I guess we are stuck with the mkdir_p for now.

Actually that's probably my argument for `.keep`ing tmp. Without it, every script would have to do that check (and no one really do it, plenty of things just assume it exists). IMO, it should be like `logs`, you should always be able to write to it without having to worry about it not being there.